pkg/controller/operators/catalog/scratch.go

```diff
@@ -0,0 +1,383 @@
+				catsrcQueue,
+				catsrcInformer.Informer(),
+				func(obj interface{}) (syncErr error) {
+					// TODO: implement (View -> Notify) -> Act stages.
+					// View stage: generates all views - collect aggregate error
+					// Notify: notify all interested parties by looking at the indexes (only if they have changed) - collect aggregate error
+					// Act stage: uses views to act on the cluster - collect aggregate error
+					// View and Notify should be synchronous
+					// Actions can execute in parallel to (View and Notify) and possibly to themselves
+					// Preferrably these stages are encapsulated by a QueueInformer wrapper... ViewInformer?
+					return
+				},
+
+				// TODO: replace nil with view cleanup
+				nil,
+				queueName,
+				metrics.NewMetricsNil(),
+				op.logger))
+
+		// Pod create pod queue informers
+		factory := informers.NewSharedInformerFactoryWithOptions(op.OpClient.KubernetesInterface(), op.resyncPeriod, informers.WithNamespace(namespace))
+		podInformer := factory.Core().V1().Pods()
+		lister.CoreV1().RegisterPodLister(namespace, podInformer.Lister())
+		// TODO: Use MetaView as a mechanism to notify CatalogSource changes
+	}
+
+	return op, nil
+}
+
+// Define
+
+// Define views
+
+// metaView is a transparent view that can be used as a basis for more complex views.
+// It holds a set of commonly used utilities.
+// TODO: should this just be the Operator itself?
+type metaView struct {
+	logger       *logrus.Logger
+	opClient     operatorclient.ClientInterface
+	lister       operatorlister.OperatorLister
+	getReference func(runtime.Object) (*corev1.ObjectReference, error)
+	now          func() metav1.Time
+}
+
+func (view *metaView) Key(value interface{}) (key string, err error) {
+	// Use the most common key func
+	// TODO: could we use metaView to store anything that implements meta.Interface by default?
+	return cache.DeletionHandlingMetaNamespaceKeyFunc(value)
+}
+
+func (view *metaView) Value(obj interface{}) (value interface{}, err error) {
+	// Passthrough
+	value = obj
+	return
+}
+
+type viewOption func(View)
+
+func applyViewOptions(view View, options ...viewOption) {
+	for _, option := range options {
+		option(view)
+	}
+}
+
+// NewMetaView returns a new metaView with the given options applied.
+func NewMetaView(options ...viewOption) *metaView {
+	// Set defaults
+	view := &metaView{
+		lister:       operatorlister.NewLister(),
+		getReference: operators.GetReference,
+		now:          timeNow,
+	}
+
+	applyViewOptions(view, options...)
+
+	// Set additional defaults if not updated by options
+	if view.opClient == nil {
+		view.opClient = operatorclient.NewClientFromConfig("", view.logger)
+	}
+
+	return view
+}
+
+// subCatalogView implements the SubscriptionCatalogStatus view over CatalogSources.
+type subCatalogView struct {
+	*metaView
+	reconcilerFactory reconciler.RegistryReconcilerFactory
+}
+
+func NewSubCatalogView(configmapRegistryImage string, options ...viewOption) *subCatalogView {
```

Comment:
nit, but maybe we want to call these `ViewBuilders` or `Viewers` and call the output of `.Value()` be the `View` object?

Reply:
Definitely want better names with less overlap.